docs/contribute/plugin/develop_plugin_cpp.md

```diff
@@ -179,58 +219,58 @@ Plugin::PluginRegister WasmEdgePluginTestEnv::Register(&Descriptor);
 } // namespace WasmEdge
 ```
 
-## Build
+Remember to implement any additional functions or structures your plug-in requires to fulfill its functionality.
 
-To build the plug-in shared library, developers should build in cmake with the WasmEdge source.
+Following these steps and implementing the necessary functions and descriptors, you can create a plug-in with host functions and modules in WasmEdge C++ API. You can continue developing your plug-in by adding functionality and implementing the desired behavior.
 
-Assume that the folder `test` is created under the `<PATH_TO_WASMEDGE_SOURCE>/plugins`.
+## Build plug-in
 
-Add this line in the `<PATH_TO_WASMEDGE_SOURCE>/plugins/CMakeLists.txt`:
+To build the plug-in shared library, developers should build in CMake with the WasmEdge source.
 
-```cmake
-add_subdirectory(test)
-```
+- Assume that the' test' folder is created under the `<PATH_TO_WASMEDGE_SOURCE>/plug-ins`. Add this line in the `<PATH_TO_WASMEDGE_SOURCE>/plugins/CMakeLists.txt`:
 
-Copy the `testplugin.h` and `testplugin.cpp` into the `<PATH_TO_WASMEDGE_SOURCE>/plugins/test` directory.
+  ```cmake
+  add_subdirectory(test)
+  ```
 
-And then edit the file `<PATH_TO_WASMEDGE_SOURCE>/plugins/test/CMakeLists.txt`:
+- Copy the `testplugin.h` and `testplugin.cpp` into the `<PATH_TO_WASMEDGE_SOURCE>/plugins/test` directory. And then edit the file `<PATH_TO_WASMEDGE_SOURCE>/plugins/test/CMakeLists.txt`:
 
-```cmake
-wasmedge_add_library(wasmedgePluginTest
-  SHARED
-  testplugin.cpp
-)
-
-target_compile_options(wasmedgePluginTest
-  PUBLIC
-  -DWASMEDGE_PLUGIN
-)
+  ```cmake
+  wasmedge_add_library(wasmedgePluginTest
```

Comment:
Please refer to the following file to update the cmake code block (except the library name):
https://github.com/WasmEdge/WasmEdge/blob/master/plugins/wasi_logging/CMakeLists.txt